03_analysis/reader.py

```diff
@@ -0,0 +1,292 @@
+        #mlist = mlist[mol_slice[0]:mol_slice[1]:mol_slice[2]]
+
+        def incrementer(count, mols, step):
+            if step == 1:
+                count += 1
+                return count
+            # use step-1 because for loop already increments once
+            for j in range(step-1):
+                count += 1
+                next(mols)
+            return count
+
+        mlist = []
+        count = 0
+        for i, m in enumerate(mols):
+
+            if count >= mol_slice[1]:
+                return mlist
+            elif count < mol_slice[0]:
+                count += 1
+                continue
+            else:
+                # important to append copy else still linked to orig generator
+                mlist.append(copy.copy(m))
+                try:
+                    count = incrementer(count, mols, mol_slice[2])
+                except StopIteration:
+                    return mlist
+
+        return mlist
+
+    return mols
+
+
+def get_sd_list(mol, taglabel):
+    """
+    Get list of specified SD tag for all confs in mol.
+
+    Parameters
+    ----------
+    mol : OEMol with N conformers
+    taglabel : string
+        tag from which to extract SD data
+
+    Returns
+    -------
+    sdlist : list
+        N-length list with value from SD tag
+
+    """
+
+    sd_list = []
+
+    for j, conf in enumerate(mol.GetConfs()):
+        for x in oechem.OEGetSDDataPairs(conf):
+            if taglabel.lower() in x.GetTag().lower():
+                sd_list.append(x.GetValue())
+                break
+
+    return sd_list
+
+def read_check_input(infile):
+    """
+    Read input file into an ordered dictionary.
+
+    Parameters
+    ----------
+    infile : string
+        name of input file to match script
+
+    Returns
+    -------
+    in_dict : OrderedDict
+        dictionary from input file, where key is method and value is dictionary
+        first entry should be reference method
+        in sub-dictionary, keys are 'sdfile' and 'sdtag'
+
+    """
+
+    in_dict = collections.OrderedDict()
+
+    # read input file
+    with open(infile) as f:
+        for line in f:
+
+            # skip commented lines or empty lines
+            if line.startswith('#'):
+                continue
+            dataline = [x.strip() for x in line.split(',')]
+            if dataline == ['']:
+                continue
+
+            # store each file's information in dictionary of dictionaries
+            in_dict[dataline[0]] = {'sdfile': dataline[1], 'sdtag': dataline[2]}
+
+    # check that each file exists before starting
+    list1 = []
+    for vals in in_dict.values():
+        list1.append(os.path.isfile(vals['sdfile']))
+
+    # if all elements are True then all files exist
+    if all(list1):
+        return in_dict
+    else:
+        print(list1)
+        raise ValueError("One or more listed files not found")
+
+
+def rdmol_from_oemol(oemol):
```

Comment:
The OFF toolkit may be able to do this for you (though you should check that it copies the data you want, the way you want) -- Consider experimenting with 
```
from openforcefield.topology import Molecule
rdmol = Molecule.from_openeye(oemol).to_rdkit()
```

Reply:
I appreciate the tip! This would reduce the amount of code I need (esp as this function was reused from another project). Right now it's giving me "unspecified stereochemistry" issues for which the current approach handles the molecules fine. Maybe related to this [OFF issue](https://github.com/openforcefield/openforcefield/issues/412). Hopefully can incorporate this function down the line.